src/webapp/github.py

```diff
@@ -88,35 +151,202 @@ def hit_merge_button(self, owner, repo, num, sha, title=None, msg=None):
+                    author: GithubUser = None):
+        """https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#create-or-update-file-contents"""
+        api_path = "/repos/:owner/:repo/contents/:path"
+        url = api_path2url(api_path, owner=owner, repo=repo, path=path)
+        data = {
+            "message": message,
+            "content": content
+        }
+        if sha:        data['sha'] = sha
+        if branch:     data['branch'] = branch
+        if committer:  data['committer'] = committer.to_dict()
+        if author:     data['author'] = author.to_dict()
+
+        return self.github_api_call("PUT", url, data)
+
+    def create_pull(self, owner, repo, title, head, base, body=None, maintainer_can_modify: bool = None,
+                    draft: bool = None, issue: int = None):
+        """https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#create-a-pull-request"""
+        api_path = "/repos/:owner/:repo/pulls"
+        url = api_path2url(api_path, owner=owner, repo=repo)
+        data = {
+            "head": head,
+            "base": base
+        }
+        if title: data['title'] = title
+        if body: data['body'] = body
+        if maintainer_can_modify: data['maintainer_can_modify'] = maintainer_can_modify
+        if draft: data['draft'] = draft
+        if issue: data['issue'] = issue
+
+        return self.github_api_call("POST", url, data)
+
+    def get_repo(self, owner, repo):
+        """https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#get-a-repository"""
+        api_path = "/repos/:owner/:repo"
+        url = api_path2url(api_path, owner=owner, repo=repo)
+
+        return self.github_api_call("GET", url, None)
+
+    def get_user(self):
+        """https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#get-the-authenticated-user"""
+        api_path = "/user"
+        url = api_path2url(api_path)
+
+        return self.github_api_call("GET", url, None)
+
+    def get_user_email(self):
+        """https://docs.github.com/en/rest/users/emails?apiVersion=2022-11-28#list-email-addresses-for-the-authenticated-user"""
+        api_path = "/user/emails"
+        url = api_path2url(api_path)
+
+        return self.github_api_call("GET", url, None)
+
     def get_api_url(self, url):
         return self.github_api_call('GET', url, None)
 
-    def target_repo(self, owner, repo):
-        return GitHubRepoAPI(self, owner, repo)
 
 class GitHubRepoAPI:
     # wrapper around GitHubAuth with (owner,repo) specified up front
 
-    def __init__(self, ghauth, owner, repo):
+    def __init__(self, ghauth: GitHubAuth, owner, repo):
         self.ghauth = ghauth
         self.owner = owner
         self.repo = repo
 
+        # Dump in the repos properties for easy access
+        repo_data = http_to_dict(*self.get_repo())
+        for key, value in repo_data.items():
+            setattr(self, key, value)
```

Comment:
I'm really not wild about setting object attributes to whatever is returned by the request.  Not only does it make it less clear what you're expecting to see from the response, it also seems like a security problem because you are potentially overriding method names here too.

I suggest you make some kind of setup method that you explicitly call, which does `self.repo_info = self.get_repo()` or similar.